elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java

```diff
@@ -159,18 +185,18 @@ public void inject(Object entity) {
     @Bean
     @ConditionalOnMissingBean
     @ConditionalOnProperty(name = "elide.aggregation-store.enabled", havingValue = "true")
-    public QueryEngine buildQueryEngine(EntityManagerFactory entityManagerFactory,
+    public QueryEngine buildQueryEngine(DataSource defaultDataSource,
             ObjectProvider<ElideDynamicEntityCompiler> dynamicCompiler, ElideConfigProperties settings)
             throws ClassNotFoundException {
 
-        MetaDataStore metaDataStore = null;
         if (isDynamicConfigEnabled(settings)) {
-            metaDataStore = new MetaDataStore(dynamicCompiler.getIfAvailable());
+            MetaDataStore metaDataStore = new MetaDataStore(dynamicCompiler.getIfAvailable());
+            return new SQLQueryEngine(metaDataStore, defaultDataSource, defaultDialect,
```

Comment:
Instead of 2 return statements, We can have 1 return. In this block, we can populate the Map else the default Empty map is passed.